File: Projects/learn-spring-framework/src/main/java/com/feeeeel/learnspringframework/App02HelloWorldSpring.java
package com.feeeeel.learnspringframework;

import org.springframework.context.annotation.AnnotationConfigApplicationContext;

public class App02HelloWorldSpring {
    public static void main(String[] args) {
        // 1: Launch a Spring Context
        var context = new AnnotationConfigApplicationContext(HelloWorldConfiguration.class);
        // 2: Configure the things that we want Spring to manage -
        // HelloWorldConfiguration - @Configuration
        // name - @Bean

        // 3: Retrieving Beans managed by Spring
        System.out.println(context.getBean("name"));
        System.out.println(context.getBean("age"));
        System.out.println(context.getBean("person"));
        System.out.println(context.getBean("person2MethodCall"));
        System.out.println(context.getBean("person3Parameters"));
        System.out.println(context.getBean("address2"));
//        System.out.println(context.getBean(Address.class)); // 오류: 같은 Bean(Address Class Object)이 2개 이상 있기 때문(address2, address3)
    }
}

File: Projects/learn-spring-framwork-02-kotlin/src/main/kotlin/com/feeeeel/learnspringframwork01kotlin/helloworld/App02HelloWorldSpring.kt
package com.feeeeel.learnspringframwork01kotlin.helloworld

import org.springframework.context.annotation.AnnotationConfigApplicationContext
import java.util.*

class App02HelloWorldSpring {}

fun main() {
    AnnotationConfigApplicationContext(HelloWorldConfiguration::class.java).use { context ->
        println(context.getBean("name"))
        println(context.getBean("age"))
        println(context.getBean("person"))
        println(context.getBean("person2MethodCall"))
        println(context.getBean("person3Parameters"))
        println(context.getBean("address2"))
        println(context.getBean("address3"))
        println(context.getBean(Person::class.java))
        println(context.getBean(Address::class.java))
        Arrays.stream(context.beanDefinitionNames).forEach(::println)
    }
}